### ISSUE TITLE
Configure user OAUTH / GCP Firebase Auth

### ISSUE THREAD

**Issue by Jonathan-Fort** [Jonathan-Fort/IGAIR-Web#3]
We want to be able to use OAuth for users, or for the potential for them to have a username and password that we DONT directly store. This reduces the risk for us as a platform and keeps authentication within proven and well-tested frameworks that can handle the variable load.

**Comment by Unikore** [comment]
Is the idea to hook up the site as a "Discord App" so people can log in using their Discord identity? And maybe in the future RSI if they ever get to that stage, I guess :)

**Comment by Jonathan-Fort** [comment]
If we can use Discord OAuth, that'd be preferred because we can verify org membership if we want to give people access to authentication-restricted areas of the site. 

If RSI provides a means to authenticate our users then that'll be 💯

**Comment by Unikore** [comment]
Yes, you can setup Discord Apps for your server to authenticate on any OAuth service. See here: https://discordapp.com/developers/docs/topics/oauth2

**Comment by Unikore** [comment]
Firebase Authentication only supports Discord as a "Custom OAuth Provider". Meaning, one has to provide a JWT token, there's no other way. Now, Discord actually does not provide JWTs, so I think the only way is to generate a JWT after a user has successfully signed in.

So, we'll have to do it like this: https://stackoverflow.com/a/54010742
